src/EditModel/SharingNotification.component.js

```diff
@@ -1,21 +1,21 @@
 import React from 'react';
-import Auth from 'd2-ui/lib/auth/Auth.mixin';
 import Translate from 'd2-ui/lib/i18n/Translate.mixin';
 import { config } from 'd2/lib/d2';
 import FontIcon from 'material-ui/FontIcon/FontIcon';
 import Paper from 'material-ui/Paper/Paper';
+import { withAuth } from "../utils/Auth";
 
 const SharingNotification = React.createClass({
     propTypes: {
         modelType: React.PropTypes.string,
         style: React.PropTypes.object,
     },
 
-    mixins: [Auth, Translate],
+    mixins: [Translate],
 
     render() {
-        const createPublic = this.getCurrentUser().canCreatePublic(this.getModelDefinitionByName(this.props.modelType));
-        const createPrivate = this.getCurrentUser().canCreatePrivate(this.getModelDefinitionByName(this.props.modelType));
+        const createPublic = this.props.getCurrentUser().canCreatePublic(this.props.getModelDefinitionByName(this.props.modelType));
+        const createPrivate = this.props.getCurrentUser().canCreatePrivate(this.props.getModelDefinitionByName(this.props.modelType));
```

Comment:
Please extract `this.props.getModelDefinitionByName(this.props.modelType)` to a variable to reduce the line length, reduce code duplication, and make it a bit easier to read.